src/solver/optimisation/opt_rename_problem.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+#include "opt_structure_probleme_a_resoudre.h"
+#include "opt_export_structure.h"
+
+const std::string SEPARATOR = "::";
+const std::string AREA_SEP = "$$";
+
+std::string BuildName(const std::string& name,
+                      const std::string& location,
+                      const std::string& timeIdentifier);
+
+class TargetVectorUpdater
+{
+public:
+    TargetVectorUpdater(bool isRenamingProcessed, std::vector<std::string>& target) :
+     target_(target)
+    {
+        if (isRenamingProcessed)
+        {
+            UpdateTargetAtIndex = std::bind(
+              &TargetVectorUpdater::UpdateTargetAtIndexImpl, this, std::placeholders::_1, std::placeholders::_2);
+        }
+    }
+
+    std::function<void(const std::string&, unsigned index)> UpdateTargetAtIndex
+      = [](const std::string&, unsigned index) {};
+
+    std::string& StringAtIndex(unsigned index) const
+    {
+        return target_[index];
+    }
+
+private:
+    std::vector<std::string>& target_;
+    void UpdateTargetAtIndexImpl(const std::string& full_name, unsigned index)
+    {
+        target_[index] = full_name;
+    }
+};
+
+struct CurrentAssetsStorage
+{
+    int timeStep = 0;
+    std::string origin;
+    std::string destination;
+    std::string area;
+};
+
+class Namer
+{
+public:
+    Namer(std::vector<std::string>& target, bool namedProblems) :
+     name_updater(namedProblems, target)
+    {
+    }
+
+    void UpdateTimeStep(int timeStep)
+    {
+        currentAssetsStorage.timeStep = timeStep;
+    }
+
+    void UpdateArea(const std::string& area)
+    {
+        currentAssetsStorage.area = area;
+    }
+
+    TargetVectorUpdater name_updater;
```

Comment:
`TargetVectorUpdater name_updater;` : should be renamed into : 
`TargetVectorUpdater targetUpdater_;